common_utils/src/main/java/com/storyous/commonutils/Toaster.kt

```diff
@@ -0,0 +1,251 @@
+package com.storyous.commonutils
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.res.Resources
+import android.os.Build
+import android.view.Display
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import android.widget.Toast
+import androidx.annotation.StringRes
+import timber.log.Timber
+
+object Toaster {
+
+    @JvmStatic
+    @JvmOverloads
+    fun showLong(context: Context, resId: Int, applyBlock: ((Toast) -> Unit) = {}): Toast {
+        return show(context, resId, Toast.LENGTH_LONG, applyBlock)
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun showShort(context: Context, resId: Int, applyBlock: ((Toast) -> Unit) = {}): Toast {
+        return show(context, resId, Toast.LENGTH_SHORT, applyBlock)
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun showLong(context: Context, text: String, applyBlock: ((Toast) -> Unit) = {}): Toast {
+        return show(context, text, Toast.LENGTH_LONG, applyBlock)
+    }
+
+    @JvmStatic
+    @JvmOverloads
+    fun showShort(context: Context, text: String, applyBlock: ((Toast) -> Unit) = {}): Toast {
+        return show(context, text, Toast.LENGTH_SHORT, applyBlock)
+    }
+
+    @JvmStatic
+    fun show(context: Context, text: String, duration: Int): Toast {
+        return show(context, text, duration, {})
+    }
+
+    @SuppressLint("ToastDetector")
+    fun makeText(context: Context, text: String, length: Int): Toast {
+        return if (Build.VERSION.SDK_INT == Build.VERSION_CODES.N_MR1) {
+            ToastCompat.makeText(context, text, length)
+        } else {
+            Toast.makeText(context, text, length)
+        }
+    }
+
+    private fun show(
+        context: Context,
+        resId: Int,
+        duration: Int,
+        applyBlock: ((Toast) -> Unit) = {}
+    ) = show(context, context.resources.getText(resId).toString(), duration, applyBlock)
+
+
+    private fun show(
+        context: Context,
+        text: String,
+        length: Int,
+        applyBlock: ((Toast) -> Unit) = {}
+    ): Toast {
+        Timber.i("Showing toast in $context: $text")
+        return makeText(context, text, length).apply {
+            applyBlock.invoke(this)
+            show()
+        }
+    }
+
+
+    /**
+     * @author Niharika.Arora
+     */
+    @Suppress("TooManyFunctions")
+    // TODO Remove after min SDK > 25
+    private class ToastCompat(context: Context, private val toast: Toast) : Toast(context) {
```

Comment:
Musí to být `private`? Nedalo by se to dát do jinýho souboru jako `internal`?

Reply:
urcite da, ale ja to chtel bych vsechno pohromade, kdyz se to pouzivat jen v tom Toasteru.

Reply:
radši čitelnost, prosíííím 🙏